src/main/java/com/pagsestagio/movieapi/controller/FilmeController.java

```diff
@@ -24,53 +24,53 @@ public FilmeController(FilmeService filmeService){
+            respostaRequisicao = ResponseEntity.ok().body(new FilmeRespostaRetornaFilmeOuMensagem(retornoService.idpublico(), retornoService.nomeFilme(), null));
         } else {
-            respostaRequisicao = ResponseEntity.badRequest().body(new FilmeRespostaRetornaListaDeFilmesOuExcecao(null, retornoService.mensagemStatus()));
+            respostaRequisicao = ResponseEntity.badRequest().body(new FilmeRespostaRetornaFilmeOuMensagem(null, null, retornoService.mensagemStatus()));
         }
         return respostaRequisicao;
     }
 
     @PutMapping
     public ResponseEntity<FilmeResposta> atualizarFilme(@RequestBody FilmeDTO filme) {
-        FilmeResultadoRetornaListaDeFilmesOuExcecao retornoService = filmeService.atualizarFilme(filme);
+        FilmeResultadoRetornaFilmeOuMensagem retornoService = filmeService.atualizarFilme(filme);
         ResponseEntity<FilmeResposta> respostaRequisicao = null;
         if(retornoService.mensagemStatus() == null){
-            respostaRequisicao = ResponseEntity.ok().body(new FilmeRespostaRetornaListaDeFilmesOuExcecao(retornoService.listaDeFilmes(), null));
+            respostaRequisicao = ResponseEntity.ok().body(new FilmeRespostaRetornaFilmeOuMensagem(retornoService.idpublico(), retornoService.nomeFilme(), null));
         } else {
-            respostaRequisicao = ResponseEntity.badRequest().body(new FilmeRespostaRetornaListaDeFilmesOuExcecao(null, retornoService.mensagemStatus()));
+            respostaRequisicao = ResponseEntity.badRequest().body(new FilmeRespostaRetornaFilmeOuMensagem(null, null, retornoService.mensagemStatus()));
         }
         return respostaRequisicao;
     }
 
-    @DeleteMapping("/{id}")
-    public ResponseEntity<FilmeResposta> deletarFilmePeloId(@PathVariable Integer id){
-        FilmeResultadoRetornaListaDeFilmesOuExcecao retornoService = filmeService.deletarFilmePeloId(id);
+    @DeleteMapping("/{idpublico}")
+    public ResponseEntity<FilmeResposta> deletarFilmePeloId(@PathVariable UUID idpublico){
+        FilmeResultadoRetornaFilmeOuMensagem retornoService = filmeService.deletarFilmePeloId(idpublico);
         ResponseEntity<FilmeResposta> respostaRequisicao = null;
 
-        if(retornoService.mensagemStatus() == null){
-            respostaRequisicao = ResponseEntity.ok().body(new FilmeRespostaRetornaListaDeFilmesOuExcecao(retornoService.listaDeFilmes(), null));
+        if(retornoService.mensagemStatus().equals("Filme excluído com sucesso!")){
+            respostaRequisicao = ResponseEntity.ok().body(new FilmeRespostaRetornaFilmeOuMensagem(null, null, retornoService.mensagemStatus()));
         } else {
-            respostaRequisicao = ResponseEntity.status(HttpStatus.NOT_FOUND).body(new FilmeRespostaRetornaListaDeFilmesOuExcecao(null, retornoService.mensagemStatus()));
+            respostaRequisicao = ResponseEntity.status(HttpStatus.NOT_FOUND).body(new FilmeRespostaRetornaFilmeOuMensagem(null, null, retornoService.mensagemStatus()));
```

Comment:
talvez para você treinar um pattern, seria legal tentar usar aqui um factory
https://refactoring.guru/pt-br/design-patterns/factory-method

pode ser nesse pr.
ou se quiser, podemos colocar numa próxima task.

a ideia seria 

seria criar um método que encapsula essa criação de objetos.

algo como 


```
public FilmeResposta factoryFilmeResposta(FilmeResultadoRetornaFilmeOuMensagem retornoService, String httpMethod)  {
 FilmeResposta resposta;
  if(httpMethod == "DELETE") {
       resposta = new FilmeRespostaRetornaFilmeOuMensagem(null, null, retornoService.mensagemStatus()))
  } else if(retornaService.mensagem != null && httpMethod == "GET")  {
     ...
  } else if..
 ...
   return resposta;


}
```

Reply:
a ideia ser encapsular a criação e as regras de criação de um objeto numa classe/método.

Reply:
Não entendi como funcionaria esse factory.

Reply:
o factory é um padrão. um padrão que usamos para criar objetos.
um padrão criacional.
https://refactoring.guru/pt-br/design-patterns/creational-patterns

tipo, repara que quando você vai criar um objeto tem um conjuntinho de lógica.
ah se o campo x for igual a y então crio desse jeito se o campo x for igual a z eu crio desse outro jeito.
Se caso  estiver dentro do do controller de get eu passo esses valores para criar o objeto e crio o objeto de Get
Se caso estiver dentro do controler de post eu passo outros valores para o criar o objeto e crio o objeto de Post.


A ideia é ter um classe que faz tudo isso. (tipo aquele helper)
que encapsula essa lógica e oferece uma api simples de uma usar 
(um método que você passa os argumentos que tem todos esses if e else necessario para criar o objeto que você precisa.)

Reply:
mas como eu vi que você ficou com dúvida colocar numa próxima task